### PR TITLE
Handle more time formats and allow custom formats

### LINKED ISSUE #40: Handle more input types for the Time type
Especially for the date time helpers in HTML. That way they work out of the box

### PR DESCRIPTION
**Pull request by paulcsmith**
Closes #40

### CODE REVIEW

File: src/avram/charms/time_extensions.cr [hunk 1]
```diff
@@ -3,16 +3,51 @@ struct Time
...
+      end.try do |format|
+        SuccessfulCast.new format.parse(value).to_utc
+      end
+    end
+
+    def self.try_parsing_with_string_formats(value)
+      Avram.settings.time_formats.find do |format|
+        begin
+          Time.parse(value, format, Time::Location.load("UTC"))
+        rescue e : Time::Format::Error
```
Comment (paulcsmith) [inline src/avram/charms/time_extensions.cr]: I wish Crystal had some way to return `nil`  but it only raises when parsing so we have to do this kind of hacky begin/rescue dance :(

Comment (jwoertink) [reply src/avram/charms/time_extensions.cr]: Working with Time in crystal is actually really clunky.

**Comment by jwoertink** [comment]
Looks good to me!

**Comment by paulcsmith** [comment]
Thanks for the reviews. And thanks for the kind words Edward!

**Pull request merged**